hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmKeyLocationInfoGroup.java

```diff
@@ -37,38 +37,59 @@ public class OmKeyLocationInfoGroup {
   private final Map<Long, List<OmKeyLocationInfo>> locationVersionMap;
   private  boolean isMultipartKey;
 
-  public OmKeyLocationInfoGroup(long version,
-                                List<OmKeyLocationInfo> locations) {
-    this(version, locations, false);
-  }
-
-  public OmKeyLocationInfoGroup(long version,
-      List<OmKeyLocationInfo> locations, boolean isMultipartKey) {
-    this.version = version;
-    locationVersionMap = new HashMap<>();
-    for (OmKeyLocationInfo info : locations) {
-      locationVersionMap
-          .computeIfAbsent(info.getCreateVersion(), v -> new ArrayList<>())
-          .add(info);
+  public OmKeyLocationInfoGroup(Builder objBuilder) {
+    this.version = objBuilder.version;
+    if (objBuilder.mapLocations != null) {
+      this.locationVersionMap = objBuilder.mapLocations;
+    } else {
+      locationVersionMap = new HashMap<>();
+      for (OmKeyLocationInfo info : objBuilder.listLocations) {
+        locationVersionMap
+            .computeIfAbsent(info.getCreateVersion(), v -> new ArrayList<>())
+            .add(info);
+      }
     }
     //prevent NPE
-    this.locationVersionMap.putIfAbsent(version, new ArrayList<>());
-    this.isMultipartKey = isMultipartKey;
-
+    this.locationVersionMap.putIfAbsent(objBuilder.version, new ArrayList<>());
+    this.isMultipartKey = objBuilder.isMultipartKey;
   }
 
-  public OmKeyLocationInfoGroup(long version,
-                                Map<Long, List<OmKeyLocationInfo>> locations) {
-    this(version, locations, false);
-  }
+  /**
+   * Builder Class for OmKeyLocationInfoGroup.
+   */
+  public static class Builder {
+    private Map<Long, List<OmKeyLocationInfo>> mapLocations;
+    private boolean isMultipartKey = false;
+    private List<OmKeyLocationInfo> listLocations;
+    private long version;
+
+    public Builder setMapLocations(
+        Map<Long, List<OmKeyLocationInfo>> mapLocations) {
+      this.mapLocations = mapLocations;
+      return this;
+    }
 
-  public OmKeyLocationInfoGroup(long version,
-      Map<Long, List<OmKeyLocationInfo>> locations, boolean isMultipartKey) {
-    this.version = version;
-    this.locationVersionMap = locations;
-    //prevent NPE
-    this.locationVersionMap.putIfAbsent(version, new ArrayList<>());
-    this.isMultipartKey = isMultipartKey;
+    public Builder setListLocations(List<OmKeyLocationInfo> listLocations) {
+      this.listLocations = listLocations;
+      return this;
+    }
+
+    public Builder setIsMultipartKey(boolean isMultipartKey) {
+      if (isMultipartKey) {
+        this.isMultipartKey = true;
+      }
+      return this;
+    }
+
+    public Builder setVersion(long version) {
+      this.version = version;
+      return this;
+    }
+
+    //Return the final constructed builder object
+    public OmKeyLocationInfoGroup build() {
+      return new OmKeyLocationInfoGroup(this);
```

Comment:
One advantage of builders is that they can validate the consistency of the object being built.  For example, `OmKeyLocationInfoGroup` should be constructed either with map or list, but not both.  The existing code ensures this, since there are no constructors with both parameters.  However, the new builder allows setting both.  Silently preferring one or the other is error-prone.